project/src/com/google/daggerquery/executor/models/Query.java

```diff
@@ -308,12 +302,50 @@ private void checkNodeForCorrectness(String node, BindingGraph bindingGraph) {
     List<String> closestNodes = new ArrayList<>();
 
     for (String node: allNodes) {
-      int distance = levenshteinDistanceCalculator.apply(originalNode, node);
+      int distance = calculateDistance(originalNode, node);
       if (distance <= MAX_NUMBER_OF_MISPLACED_LETTERS) {
         closestNodes.add(node);
       }
     }
 
     return closestNodes;
   }
+
+  /**
+   * Measures the difference between two strings.
+   *
+   * @return the number of changes required to get one string from another,
+   * where each change is a single character modification (substitution, deletion or insertion)
+   */
+  private int calculateDistance(String firstString, String secondString) {
```

Comment:
Should we make this algorithm case insensitive?

Reply:
I'm not sure - since nodes names are case sensitive, it makes sense to distinguish `com.google.A` from `com.google.a`.

Reply:
You're right :)